File: CSharp/Swim.Data/Repositories/DatabaseRepository.cs
using log4net;
using Swim.Data.Utils;
using Swim.Domain.Utils;

namespace Swim.Data.Repositories
{
    public abstract class DatabaseRepository
    {
        protected readonly DatabaseConnection _connection = DatabaseConnection.Instance;
        protected static readonly ILog _logger = LogManager.GetLogger(typeof(DatabaseRepository));
    }
}

File: Java/src/main/java/swim/Data/Repositories/DatabaseRepository.java
package swim.Data.Repositories;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import swim.Data.Utils.DatabaseConnection;
import swim.Domain.Utils.Entity;

public abstract class DatabaseRepository<ID, E extends Entity<ID>> implements Repository<ID, E>{
    protected static DatabaseConnection connection = DatabaseConnection.getInstance();
    protected static Logger logger = LogManager.getLogger("DatabaseRepository");
}
